File: projects/08/java/TemperatureTester.java
/* NameTester.java is a 'driver' to test class Name.
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Student: Jacob Brink
 * Date: 3/20/2020
 * Project: 08
 ************************************************************/

import java.io.*;
import java.util.Scanner;


public class TemperatureTester  
{

    public static Scanner keyboard;
    
    public static void main(String[] args)
    {
	keyboard = new Scanner(System.in);
	Temperature baseTemp, limitTemp;

	//get first temperature
	System.out.println("Enter baseTemp: ");
	try {
	    baseTemp = Temperature.read(keyboard);
	} catch (Exception e) {
	    System.out.println("Invalid Temperature");
	    return;
	}

	//get limit temperature
	System.out.println("Enter limitTemp: ");
	try {
	    limitTemp = Temperature.read(keyboard);
	} catch (Exception e) {
	    System.out.println("Invalid Temperature");
	    return;
	}

	
	System.out.println(String.format("Enter stepValue in scale %s", baseTemp.getScale()));
	double stepValue = keyboard.nextDouble();

	//loop either up or down depending on whether or not stepValue is negative or positive
	while ((stepValue > 0 ? Temperature.lessThan(baseTemp, limitTemp) : Temperature.lessThan(limitTemp, baseTemp)) || baseTemp.equals(limitTemp)) {
	    System.out.println(String.format("%8s\t%8s\t%8s", Temperature.getFahrenheit(baseTemp), Temperature.getCelsius(baseTemp), Temperature.getKelvin(baseTemp)));
	    try {
		baseTemp = baseTemp.raise(stepValue);
	    } catch (Exception e) {
		System.out.println("The change in temperature resulted in an invalid temperature");
		return;
	    };
	}
	    

    }
}

File: projects/08/clojure/src/temperatureTester.clj
;; temperatureTester.clj tests a Clojure Temperature type,
;;  stored in Temperature.clj (in the src directory).
;;
;; Output: the results of testing the Name functions.
;;
;; Usage: clojure -m temperatureTester
;;
;; Begun by: Prof. Adams, CS 214 at Calvin College.
;; Completed by: Jacob Brink
;; Date: 4/14/2020
;; Project: 8
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns temperatureTester)


(load "Temperature")


;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; A simple driver to test our Temperature functions. 
;; Output: the table results of testing our Temperature functions.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (let
    [                                          ; 3 ways to construct an object:
     baseTemp (read-Temperature)
     limitTemp (read-Temperature)
     stepValue (Double/parseDouble (read-line))
    ]
    ;; ----- SECTION 1 -----
    (println)

    ;; if step value is positive loop up otherwise loop down

        (loop [currentTemp baseTemp]
          (if (or
               (if (> stepValue 0) (less-than currentTemp limitTemp) (less-than limitTemp currentTemp))
               (equals currentTemp limitTemp)
               )
            (do
              (print (printTemperature (getFahrenheit currentTemp)) " "
                     (printTemperature (getCelsius currentTemp)) " "
                     (printTemperature (getKelvin currentTemp)))
              (println) (flush)
              (recur (raise currentTemp stepValue))
              )
            )
          )        
      )
    
  )



